src/base/search/searchpluginmanager.cpp

```diff
@@ -122,6 +122,27 @@ QStringList SearchPluginManager::supportedCategories() const
     return result;
 }
 
+QStringList SearchPluginManager::getPluginCategories(const QString &pluginName) const
+{
+    QStringList plugins;
+    if (pluginName == "all")
+        plugins = allPlugins();
+    else if ((pluginName == "enabled") || (pluginName == "multi"))
+        plugins = enabledPlugins();
+    else
+        plugins << pluginName.trimmed();
+
+    QSet<QString> categories;
+    for (const QString &pluginName : plugins) {
```

Comment:
You should use `qAsConst` with plugins to prevent detach.
Don't forget to add `#include "base/global.h"`

Reply:
Added `qAsConst`